lib/asciidoctor/document.rb

```diff
@@ -1047,11 +1047,12 @@ def convert opts = {}
     #end
 
     if doctype == 'inline'
-      # QUESTION should we warn if @blocks.size > 0 and the first block is not a paragraph?
-      if (block = @blocks[0]) && block.content_model != :compound
-        output = block.content
-      else
-        output = nil
+      if (block = @blocks[0])
+        if block.content_model == :compound || block.content_model == :empty
+          warn %(asciidoctor: WARNING: no inline candidate; use the inline doctype to convert a single paragragh, verbatim, or raw block)
```

Comment:
I'm not really a fan of adding links to warning messages. I think Maven tried that and it ended up being very misleading (though it could have been how they did it). I'd actually prefer the user to search because I think it leads them to very valuable information. But that doesn't mean I want to send them on a wild goose chase either. When we resolve #44, we can perhaps assign error codes that would match entries in a table in the documentation. That would get them to an explanation in the least amount of steps.